services/galley/src/Galley/Data.hs

```diff
@@ -749,25 +780,32 @@ memberLists convs = do
 members :: (MonadClient m, Log.MonadLogger m, MonadThrow m) => ConvId -> m [LocalMember]
 members conv = join <$> memberLists [conv]
 
+lookupRemoteMembers :: (MonadClient m) => ConvId -> m [RemoteMember]
+lookupRemoteMembers conv = join <$> remoteMemberLists [conv]
+
 -- | Add a member to a local conversation, as an admin.
-addMember :: MonadClient m => UTCTime -> ConvId -> UserId -> m (Event, List1 LocalMember)
+addMember :: MonadClient m => UTCTime -> ConvId -> UserId -> m (Event, [LocalMember])
 addMember t c u = addMembersUnchecked t c u (singleton u)
 
 -- | Add members to a local conversation.
-addMembersWithRole :: MonadClient m => UTCTime -> ConvId -> (UserId, RoleName) -> ConvMemberAddSizeChecked (List1 (UserId, RoleName)) -> m (Event, List1 LocalMember)
-addMembersWithRole t c orig mems = addMembersUncheckedWithRole t c orig (fromMemberSize mems)
+addMembersWithRole :: MonadClient m => UTCTime -> ConvId -> (UserId, RoleName) -> ConvMemberAddSizeChecked -> m (Event, [LocalMember], [RemoteMember])
+addMembersWithRole t c orig mems = addMembersUncheckedWithRole t c orig (sizeCheckedLocals mems) (sizeCheckedRemotes mems)
 
 -- | Add members to a local conversation, all as admins.
 -- Please make sure the conversation doesn't exceed the maximum size!
-addMembersUnchecked :: MonadClient m => UTCTime -> ConvId -> UserId -> List1 UserId -> m (Event, List1 LocalMember)
-addMembersUnchecked t conv orig usrs = addMembersUncheckedWithRole t conv (orig, roleNameWireAdmin) ((,roleNameWireAdmin) <$> usrs)
+addMembersUnchecked :: MonadClient m => UTCTime -> ConvId -> UserId -> List1 UserId -> m (Event, [LocalMember])
```

Comment:
Shouldn't this be called `addLocalMembersUnchecked`?

Reply:
Good idea.